internal/controllers/nmc_reconciler.go

```diff
@@ -198,25 +228,63 @@ func FindNodeCondition(cond []v1.NodeCondition, conditionType v1.NodeConditionTy
 
 //go:generate mockgen -source=nmc_reconciler.go -package=controllers -destination=mock_nmc_reconciler.go workerHelper
 
-type workerHelper interface {
-	ProcessModuleSpec(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig, spec *kmmv1beta1.NodeModuleSpec, status *kmmv1beta1.NodeModuleStatus) error
+type nmcReconcilerHelper interface {
+	GarbageCollectInUseLabels(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig) error
+	ProcessModuleSpec(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig, spec *kmmv1beta1.NodeModuleSpec, status *kmmv1beta1.NodeModuleStatus) (bool, error)
 	ProcessOrphanModuleStatus(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig, status *kmmv1beta1.NodeModuleStatus) error
-	SyncStatus(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig) error
 	RemoveOrphanFinalizers(ctx context.Context, nodeName string) error
+	SyncStatus(ctx context.Context, nmc *kmmv1beta1.NodeModulesConfig) error
 }
 
-type workerHelperImpl struct {
+type nmcReconcilerHelperImpl struct {
 	client client.Client
 	pm     podManager
 }
 
-func newWorkerHelper(client client.Client, pm podManager) workerHelper {
-	return &workerHelperImpl{
+func newWorkerHelper(client client.Client, pm podManager) nmcReconcilerHelper {
+	return &nmcReconcilerHelperImpl{
 		client: client,
 		pm:     pm,
 	}
 }
 
+func (h *nmcReconcilerHelperImpl) GarbageCollectInUseLabels(ctx context.Context, nmcObj *kmmv1beta1.NodeModulesConfig) error {
```

Comment:
I think that this function  could be much more easier understood if it was written like:
for inUseLabel in labels:
     if module not present in Spec and not in Status:
            add to labelsTo Delete
patch NMC with labelsToDelete

Reply:
Then lets add a comment explaining what that function does, because it takes some time to understand it

Reply:
Added a comment.